### PR TITLE
Notification Inbox

### PR DESCRIPTION
**Pull request by bweengener**

### CODE REVIEW

File: app/assets/stylesheets/notifications.scss
Comment (Durborough) [inline app/assets/stylesheets/notifications.scss:4]: Shouldn't this be put in a global CSS variable?

File: app/components/notification_inbox_component.html.erb
Comment (Durborough) [inline app/components/notification_inbox_component.html.erb:3]: Just a question here: Can't we create the `NotificationComponent`s first and then render them all at once?

File: app/mailers/notification_mailer.rb
Comment (Durborough) [inline app/mailers/notification_mailer.rb:24]: Here `Bookeeper Red Notification` has a Typo. Also: Why can't we use the I18n Stuff here?

File: app/controllers/notifications_controller.rb
Comment (Durborough) [inline app/controllers/notifications_controller.rb:7]: I think we should add a comment here why we dont delete the notification for good and instead set `    @notification.display = false`.

File: app/views/items/index.html.erb
Comment (Durborough) [inline app/views/items/index.html.erb:9]: What does this placeholder do?

File: app/views/layouts/_navbar.html.erb
Comment (Durborough) [inline app/views/layouts/_navbar.html.erb:85]: There should be an I18n fpr the Notifications lable.

File: app/views/notification_mailer/send_notification.html.erb
Comment (Durborough) [inline app/views/notification_mailer/send_notification.html.erb:1]: There should be I18n for this.

File: app/views/notification_mailer/send_notification.html.erb
Comment (Durborough) [inline app/views/notification_mailer/send_notification.html.erb:7]: Mixing english and german isn't that great. Also: This kind of shenenigans should not go into production code!

File: db/migrate/20221206220527_create_notifications.rb
Comment (Durborough) [inline db/migrate/20221206220527_create_notifications.rb:5]: Can we restrict here which kind of notification_types are accepted?

File: spec/models/notification_spec.rb
Comment (Durborough) [inline spec/models/notification_spec.rb:4]: This should not be left pending! Add tests please!

**Review by Durborough — CHANGES_REQUESTED** [review]
Looks good, but I think there are still tests missing. This time we should really add them since there is some custom logic. Also I18n isn't always used. I think we are not responsible for the failing tests, or are we?

File: app/assets/stylesheets/notifications.scss
Comment (bweengener) [reply app/assets/stylesheets/notifications.scss:4]: Feel free to do so.

File: app/components/notification_inbox_component.html.erb
Comment (bweengener) [reply app/components/notification_inbox_component.html.erb:3]: I don’t get your question. What’s the problem with this approach?

File: app/mailers/notification_mailer.rb
Comment (bweengener) [reply app/mailers/notification_mailer.rb:24]: 👍🏻

File: app/views/items/index.html.erb
Comment (bweengener) [reply app/views/items/index.html.erb:9]: Not our business. Have a look at the path. This file is listed as I did a code cleanup.

File: app/views/layouts/_navbar.html.erb
Comment (bweengener) [reply app/views/layouts/_navbar.html.erb:85]: 👍🏻

File: app/views/notification_mailer/send_notification.html.erb
Comment (bweengener) [reply app/views/notification_mailer/send_notification.html.erb:7]: 👍🏻

File: spec/components/notification_component_spec.rb
Comment (bweengener) [reply spec/components/notification_component_spec.rb:6]: @Durborough Feel free to add tests.

File: spec/components/notification_inbox_component_spec.rb
Comment (bweengener) [reply spec/components/notification_inbox_component_spec.rb:6]: Feel free to add tests.

File: spec/factories/notifications.rb
Comment (bweengener) [reply spec/factories/notifications.rb:6]: Auto-generated. A better default would be `true`.

File: app/mailers/notification_mailer.rb
Comment (gruendlich) [reply app/mailers/notification_mailer.rb:24]: It would be great to use translation here. The problem I see, however, is that in difference to displaying a page using a certain language, we here don't know which language the user prefers. I intend to look into it tomorrow.

**Comment by gruendlich** [comment]
> 

Agreed, tests are necessary this time. If not already done by tomorrow, I'll concern myself with it.

File: app/mailers/notification_mailer.rb
Comment (bweengener) [reply app/mailers/notification_mailer.rb:24]: This is a cross-cutting concern. One solution would be persisting the language-preference in the database each time the user toggles it. 

@EliasNotFound (PM) Can you request the mentioned feature to the team which was responsible for the language picker?

File: app/views/notification_mailer/send_notification.html.erb
Comment (gruendlich) [reply app/views/notification_mailer/send_notification.html.erb:7]: @Durborough Neither was it ever intended to. As stated before, I only got to work on it on Wednesday and intended and still intend to work on it on Monday / Tuesday. The creation of a pull request did come as a surprise to me.

File: db/migrate/20221206220527_create_notifications.rb
Comment (gruendlich) [reply db/migrate/20221206220527_create_notifications.rb:5]: Yes, notification type is an enum.

File: spec/mailers/previews/notification_mailer_preview.rb
Comment (gruendlich) [reply spec/mailers/previews/notification_mailer_preview.rb:15]: Why?

File: app/views/notification_mailer/send_notification.html.erb
Comment (bweengener) [reply app/views/notification_mailer/send_notification.html.erb:7]: @gruendlich The PR was not opened to mark this work as done, but to have continuous/early feedback and include @Durborough .

File: app/components/notification_inbox_component.html.erb
Comment (Durborough) [reply app/components/notification_inbox_component.html.erb:3]: I thought that one render call might be better than multiple ones?

File: app/views/layouts/_navbar.html.erb
Comment (Durborough) [reply app/views/layouts/_navbar.html.erb:85]: The question remains, where in the config files for the locales we should add the translation? In the folder landing_page? Have @WilliamBrandt or @EliasNotFound some insight here?

File: app/assets/stylesheets/notifications.scss
Comment (Durborough) [reply app/assets/stylesheets/notifications.scss]: I added a `$lightGrey` variable in the `Application.scss`.

File: app/mailers/notification_mailer.rb
Comment (Durborough) [reply app/mailers/notification_mailer.rb]: Fixed the Typo. Internationalization is still an issue.

**Comment by Durborough** [comment]
> > 
> 
> Agreed, tests are necessary this time. If not already done by tomorrow, I'll concern myself with it.
@gruendlich can we work on this together?

File: app/mailers/notification_mailer.rb
Comment (EliasNotFound) [reply app/mailers/notification_mailer.rb]: @bweengener I asked team 🌪️ and will keep you updated

**Comment by gruendlich** [comment]
Don't merge and don't pull. Currently broken. Last commit (merge) will be reverted.

**Review by Durborough — APPROVED** [review]

**Comment by Durborough** [comment]
@bweengener could you maybe try to resolve the merge conflicts? We tried for hours now. Somehow we cannot get the tests to work properly after trying to merge `dev` into our branch. This might be because there are some strange changes in the `gem` files in our branch, other changes in our branch don't seem to make any sense either, but we don't know how to deal with this at the moment. Also the `schema` file seems to be making some trouble? Please send help.

**Review by gruendlich — APPROVED** [review]
Functionality is now complete. We do however fail at merging dev into it.

**Comment by gruendlich** [comment]
After manually merging diffs from dev into hp/notification-inbox and fixing small issues resulting from the merge, the ci is now green, and this branch will be merged into dev now.

**Comment by TobiPeterG** [comment]
Hey, ich bin vom BA-Team (Stein) und für mich failen lokal 5 Tests, die mit items in Zusammenhang stehen (ich bin mir aber nicht sicher, ob das etwas mit dieser PR zu tun hat):
`rspec ./spec/features/items/show_spec.rb[1:1:3] # show item page with multiple screensizes doesn't show borrow button if src=qrcode is not present`
`rspec ./spec/features/items/show_spec.rb[1:1:4] # show item page with multiple screensizes doesn't show borrow button if src=not-qrcode`
`rspec ./spec/features/items/show_spec.rb[1:2:3] # show item page with multiple screensizes doesn't show borrow button if src=qrcode is not present`
`rspec ./spec/features/items/show_spec.rb[1:2:4] # show item page with multiple screensizes doesn't show borrow button if src=not-qrcode`
`rspec ./spec/features/notifications/destroy_spec.rb:16 # Notification inbox with JS no longer shows a notification when it is dismissed`


Die ersten 4 fails scheinen vom item team zu sein. Tritt der Fehler bei euch auch auf?

**Comment by gruendlich** [comment]
Moin,
auf dem momentanen Dev-stand kann ich den Fehler leider nicht reproduzieren. Auf welchem Branch/Commit tritt er denn auf?

**Pull request merged**
